apps/rush-lib/src/logic/deploy/DeployArchiver.ts

```diff
@@ -0,0 +1,97 @@
+import * as JSZip from 'jszip';
+import * as path from 'path';
+import { FileSystem, FileSystemStats, AlreadyExistsBehavior } from '@rushstack/node-core-library';
+import { IDeployState } from './DeployManager';
+
+// JSZip is dependant on Blob being declared.
+declare global {
+  // eslint-disable-next-line
+  type Blob = any;
+}
+export class DeployArchiver {
+  public static async createArchive(deployState: IDeployState): Promise<void> {
```

Comment:
Let's name this function `createArchiveAsync` so it's clear to callers that it's an async function.